playwright/src/main/java/com/microsoft/playwright/impl/PageImpl.java

```diff
@@ -523,6 +528,34 @@ public List<ElementHandle> querySelectorAll(String selector) {
     return withLogging("Page.querySelectorAll", () -> mainFrame.querySelectorAllImpl(selector));
   }
 
+  @Override
+  public void addLocatorHandler(Locator locator, Runnable handler) {
+    LocatorImpl locatorImpl = (LocatorImpl) locator;
+    if (locatorImpl.frame != mainFrame) {
+      throw new PlaywrightException("Locator must belong to the main frame of this page");
+    }
+    withLogging("Page.addLocatorHandler", () -> {
+      JsonObject params = new JsonObject();
+      params.addProperty("selector", locatorImpl.selector);
+      JsonObject json = (JsonObject) sendMessage("registerLocatorHandler", params);
+      int uid = json.get("uid").getAsInt();
+      locatorHandlers.put(uid, handler);
+    });
+  }
+
+  private void onLocatorHandlerTriggered(int uid) {
+    try {
+      Runnable handler = locatorHandlers.get(uid);
+      if (handler != null) {
+        handler.run();
+      }
+    } finally {
+      JsonObject params = new JsonObject();
+      params.addProperty("uid", uid);
+      sendMessageAsync("resolveLocatorHandlerNoReply", params);
```

Comment:
whats the difference between sendMessage and sendMessageAsync?

Reply:
The latter is not waiting for the response. One consequence is that it also doesn't throw.